### PR TITLE
Items index stats

### PR DESCRIPTION
**Pull request by VinceCarollo**
## What does this PR do?

- Adds spec for sorted stats on items index page 
- Adds two class methods to item model
  - gets most popular items that are active (5)
  - gets least popular items that are active (5)
- Adds stats to item index page for 5 most popular items and 5 least popular items

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -10,4 +10,22 @@ class Item < ApplicationRecord
...
+        .limit(5)
+  end
+
+  def self.least_5_bought
+    self.joins(:order_items)
+        .select('sum(quantity) AS order_quantity, items.*')
+        .where('order_items.fulfilled=true')
+        .group(:id)
+        .order('order_quantity')
+        .limit(5)
```
Comment (BrennanAyers) [inline app/models/item.rb:29]: Nice! These look not fun, but well done.

File: spec/factories/items.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 FactoryBot.define do
   factory :item do
     user { nil }
-    name { "MyString" }
+    sequence :name do |n|
+      "Cheese #{n}"
```
Comment (BrennanAyers) [inline spec/factories/items.rb:5]: 🧀

File: spec/features/items/index_spec.rb [hunk 3]
```diff
@@ -24,6 +12,32 @@
...
+      OrderItem.create!(item: @item_2, order: order_2, quantity: 3, price: 6.99, fulfilled: true)
+      OrderItem.create!(item: @item_3, order: order_3, quantity: 6, price: 11.99, fulfilled: true)
+      OrderItem.create!(item: @item_4, order: order_4, quantity: 7, price: 12.99, fulfilled: true)
+      OrderItem.create!(item: @item_5, order: order_5, quantity: 14, price: 13.99, fulfilled: true)
+      OrderItem.create!(item: @item_6, order: order_6, quantity: 5, price: 2.99, fulfilled: true)
+      OrderItem.create!(item: @item_1, order: order_7, quantity: 21, price: 9.99, fulfilled: true)
+      OrderItem.create!(item: @item_2, order: order_8, quantity: 31, price: 7.99, fulfilled: true)
+      OrderItem.create!(item: @item_3, order: order_9, quantity: 2, price: 12.99, fulfilled: false)
+      OrderItem.create!(item: @item_4, order: order_10, quantity: 3, price: 11.99, fulfilled: false)
+      OrderItem.create!(item: @item_5, order: order_11, quantity: 1, price: 21.99, fulfilled: false)
```
Comment (BrennanAyers) [inline spec/features/items/index_spec.rb:40]: Yeesh! But this will be helpful to work off of going forward as well.

**Review by BrennanAyers — APPROVED** [review]

**Comment by WHomer** [comment]
User Story 15, Items Index Page Statistics #68

**Review by WHomer — APPROVED** [review]
Looks good.  Nice work on the AR calls.

**Pull request merged**